app/webpacker/components/RegistrationsV2/RegistrationEdit/Refunds.jsx

```diff
@@ -1,97 +1,112 @@
-            <Table.Body>
-              {refunds.charges.map((refund) => (
-                <Table.Row key={refund.payment_id}>
-                  <Table.Cell>
-                    <Input
-                      labelPosition="right"
-                      type="text"
-                      placeholder={refund.amount}
-                    >
-                      <Label basic>$</Label>
-                      <input
-                        value={refundAmount}
-                        max={refund.amount}
-                        onChange={(event) => setRefundAmount(event.target.value)}
-                      />
-                    </Input>
-                  </Table.Cell>
-                  <Table.Cell>
-                    <Button
-                      onClick={() => refundMutation({
-                        competitionId,
-                        userId,
-                        paymentId: refund.payment_id,
-                        amount: refundAmount,
-                      })}
-                    >
-                      Refund Amount
-                    </Button>
-                  </Table.Cell>
-                </Table.Row>
-              ))}
-            </Table.Body>
-          </Table>
-        </Modal.Content>
-        <Modal.Actions>
-          <Button disabled={isMutating} onClick={onExit}>
-            Go Back
-          </Button>
-        </Modal.Actions>
-      </Modal>
-    )
+
+  if (refundsLoading) {
+    return <Loading />;
+  }
+
+  if (refunds.length === 0) {
+    return <Message success>All charges have been refunded</Message>;
+  }
+
+  return (
+    <>
+      <Header>Available Refunds:</Header>
+      <Table>
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell>Original Payment</Table.HeaderCell>
+            <Table.HeaderCell>Available to Refund</Table.HeaderCell>
+            <Table.HeaderCell>Refund Amount </Table.HeaderCell>
+            <Table.HeaderCell />
+          </Table.Row>
+        </Table.Header>
+        <Table.Body>
+          {refunds.map((refund) => (
+            <RefundRow
+              refund={refund}
+              refundMutation={refundMutation}
+              isMutating={isMutating}
+              userId={userId}
+              competitionId={competitionId}
+              key={refund.payment_id}
+            />
+          ))}
+        </Table.Body>
+      </Table>
+    </>
+  );
+}
+
+function RefundRow({
+  refund, refundMutation, isMutating, userId, competitionId,
+}) {
+  const [amountToRefund, setAmountToRefund] = useState(refund.ruby_amount_refundable);
+
+  return (
+    <Table.Row>
+      <Table.Cell>
+        {refund.human_amount_payment}
+      </Table.Cell>
+      <Table.Cell>
+        {refund.human_amount_refundable}
+      </Table.Cell>
+      <Table.Cell>
+        <AutonumericField
+          currency={refund.currency_code.toUpperCase()}
+          value={amountToRefund}
+          onChange={(event, { value }) => setAmountToRefund(value)}
```

Comment:
You can pass the setter directly if you use our custom `useInputState` hook instead of `useState`